File: 852__Peak_Index_in_a_Mountain_Array.py
class Solution(object):
    def peakIndexInMountainArray(self, A):
        """
        :type A: List[int]
        :rtype: int
        """
        i = 0
        while (A[i] < A[i+1]):
            i += 1
        return i

# Another Binary Search Solution


    def peakIndexInMountainArray(self, A):
            lo, hi = 0, len(A) - 1
            while lo < hi:
                mi = (lo + hi) / 2
                if A[mi] < A[mi + 1]:
                    lo = mi + 1
                else:
                    hi = mi
            return lo

File: Java/852__Peak_Index_in_a_Mountain_Array.java
class Solution {
    public int peakIndexInMountainArray(int[] A) {
        int l = 1, r = A.length - 2;
        while (l < r) {
            int mid = (l + r) / 2;
            if (A[mid - 1] < A[mid] && A[mid] > A[mid + 1])
                return mid;
            if (A[mid - 1] < A[mid])
                l = mid + 1;
            else
                r = mid - 1;
        }
        return l;
    }
}
